<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlantBlueprintDialog</class>
 <widget class="QDialog" name="PlantBlueprintDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>754</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>New Plant Blueprint</string>
  </property>
  <layout class="QGridLayout" name="s_gridLayout">
   <item row="7" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_drawLength">
     <property name="toolTip">
      <string>Initial draw length</string>
     </property>
     <property name="minimum">
      <double>0.100000000000000</double>
     </property>
     <property name="maximum">
      <double>10.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.500000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="15" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_maxDeviation">
     <property name="toolTip">
      <string>Maximum deviation of a branch. Set to 0 for a more rigid L-system</string>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
   </item>
   <item row="16" column="0" colspan="2">
    <widget class="QLabel" name="s_numNodesLabel">
     <property name="text">
      <string>Number of nodes per branch</string>
     </property>
    </widget>
   </item>
   <item row="26" column="1">
    <widget class="QPushButton" name="m_cancel">
     <property name="toolTip">
      <string>Cancel</string>
     </property>
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QLabel" name="s_blueprintNameLabel">
     <property name="text">
      <string>Blueprint name</string>
     </property>
    </widget>
   </item>
   <item row="10" column="0" colspan="2">
    <widget class="QLabel" name="s_rootRadiusLabel">
     <property name="text">
      <string>Root Radius</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QLabel" name="s_drawLengthLabel">
     <property name="text">
      <string>Draw Length</string>
     </property>
    </widget>
   </item>
   <item row="26" column="0">
    <widget class="QPushButton" name="m_create">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Create this shader</string>
     </property>
     <property name="text">
      <string>Create</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QSpinBox" name="m_maxDepth">
     <property name="toolTip">
      <string>Maximum fractal depth</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="9" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_drawAngle">
     <property name="toolTip">
      <string>Draw angle in degrees</string>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
     <property name="value">
      <double>45.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="11" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_rootRadius">
     <property name="toolTip">
      <string>Root radius</string>
     </property>
     <property name="minimum">
      <double>0.010000000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
   </item>
   <item row="13" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_decay">
     <property name="toolTip">
      <string>Decay rate. Higher values = faster decay</string>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QLabel" name="s_grammarFileLabel">
     <property name="text">
      <string>L-system grammar file</string>
     </property>
    </widget>
   </item>
   <item row="12" column="0" colspan="2">
    <widget class="QLabel" name="s_decayRateLabel">
     <property name="text">
      <string>Decay Rate</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QLabel" name="s_maxDepthLabel">
     <property name="text">
      <string>Maximum L-system Depth</string>
     </property>
    </widget>
   </item>
   <item row="23" column="0" colspan="2">
    <widget class="QLineEdit" name="m_woodTextureFilePath">
     <property name="toolTip">
      <string>Wood texture file path</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QLineEdit" name="m_blueprintName">
     <property name="toolTip">
      <string>Name of new blueprint</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>BlueprintName</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0" colspan="2">
    <widget class="QLabel" name="s_drawAngleLabel">
     <property name="text">
      <string>Draw Angle</string>
     </property>
    </widget>
   </item>
   <item row="25" column="0" colspan="2">
    <widget class="QLineEdit" name="m_leafTextureFilePath">
     <property name="toolTip">
      <string>Leaf texture file path</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QLineEdit" name="m_grammarFilePath">
     <property name="toolTip">
      <string>Path to grammar file.
Note this must end with &quot;.txt&quot;</string>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="echoMode">
      <enum>QLineEdit::Normal</enum>
     </property>
     <property name="placeholderText">
      <string>GrammarFile.txt</string>
     </property>
     <property name="fileExists" stdset="0">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="22" column="0">
    <widget class="QLabel" name="s_woodTextureFileLabel">
     <property name="text">
      <string>Wood Texture File</string>
     </property>
    </widget>
   </item>
   <item row="24" column="0" colspan="2">
    <widget class="QLabel" name="s_leafTextureFileLabel">
     <property name="text">
      <string>Leaf Texture File</string>
     </property>
    </widget>
   </item>
   <item row="14" column="0" colspan="2">
    <widget class="QLabel" name="s_maxdeviatonLabel">
     <property name="text">
      <string>Maximum deviation of a branch</string>
     </property>
    </widget>
   </item>
   <item row="17" column="0" colspan="2">
    <widget class="QSpinBox" name="m_numNodes">
     <property name="toolTip">
      <string>Number of nodes in a branch. Set to minimum for a rigid L-system</string>
     </property>
     <property name="minimum">
      <number>2</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
    </widget>
   </item>
   <item row="19" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_phototropismScaleFactor">
     <property name="toolTip">
      <string>Scale factor to grow towards the sun</string>
     </property>
     <property name="maximum">
      <double>10.000000000000000</double>
     </property>
    </widget>
   </item>
   <item row="18" column="0">
    <widget class="QLabel" name="s_phototropismScaleLabel">
     <property name="text">
      <string>Phototropism scale factor</string>
     </property>
    </widget>
   </item>
   <item row="20" column="0">
    <widget class="QLabel" name="s_gravitropismScale">
     <property name="text">
      <string>Gravitropism scale factor</string>
     </property>
    </widget>
   </item>
   <item row="21" column="0" colspan="2">
    <widget class="QDoubleSpinBox" name="m_gravitropismScaleFactor">
     <property name="toolTip">
      <string>Scale factor for branches to grow downwards due to gravity</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
